### PR TITLE
[gcu] Refactor generic_config_updater/test_vlan_interface.py

### PR DESCRIPTION
**Pull request by yaqiangz**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
Fix this case failed in M0 topo caused by hard codes about vlan ip.

#### How did you do it?
Get vlan ip based on current topo config.

#### How did you verify/test it?
Run test.

### CODE REVIEW

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -271,6 +271,12 @@ generic_config_updater:
     conditions:
       - "'t2' in topo_name"
 
+generic_config_updater/test_bgp_speaker.py:
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:274]: Let's move this part to a new PR

File: tests/generic_config_updater/test_vlan_interface.py [hunk 2]
```diff
@@ -95,10 +174,10 @@ def vlan_interface_tc1_xfail(duthost):
     ("remove", "Vlan1000", "192.168.0.1/21", "fc02:1000::2/64") REMOVE Unexist IPv6 address
     """
     xfail_input = [
-        ("add", "Vlan1000", "587.168.0.1/21", "fc02:1000::1/64"),
-        ("add", "Vlan1000", "192.168.0.1/21", "fc02:1000::xyz/64"),
-        ("remove", "Vlan1000", "192.168.0.2/21", "fc02:1000::1/64"),
-        ("remove", "Vlan1000", "192.168.0.1/21", "fc02:1000::2/64")
+        ("add", vlan_info["v4"]["name"], reg_replace(vlan_info["v4"]["prefix"], r"^\d*", "587"), vlan_info["v6"]["prefix"]),
```
Comment (Blueve) [inline tests/generic_config_updater/test_vlan_interface.py:177]: Why not use 587.168.0.1/21 directly? This change will cause the test inconsistence with doc (LN 171 - 174)

File: tests/vlan/test_vlan_ping.py [hunk 3]
```diff
@@ -52,7 +53,16 @@ def vlan_ping_setup(duthosts, rand_one_dut_hostname, ptfhost, nbrhosts, tbinfo):
     Teardown:   deletes ipv4 and ipv6 address on ptf hosts and removes routes to VM. Also removes residual static arp entries from tests
     """
     vm_host_info = {}
-    vm_name, vm_info = random.choice(nbrhosts.items())
+
+    vm_name, vm_info = None, None
+    topo_name = tbinfo["topo"]["name"]
+    for nbr_name, nbr_info in nbrhosts.items():
```
Comment (Blueve) [inline tests/vlan/test_vlan_ping.py:59]: Create a new PR for this change

**Review by Blueve — CHANGES_REQUESTED** [review]

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml (hunk 1 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml]: Fixed

File: tests/vlan/test_vlan_ping.py (hunk 3 above)
Comment (yaqiangz) [reply tests/vlan/test_vlan_ping.py]: Fixed

File: tests/generic_config_updater/test_vlan_interface.py (hunk 2 above)
Comment (yaqiangz) [reply tests/generic_config_updater/test_vlan_interface.py]: Vlan prefix len in different topo possible different, so I think dynamic replacement by regex is more flexible than directly use

**Review by Blueve — APPROVED** [review]

**Pull request merged**